lib/internal/http2/core.js

```diff
@@ -3119,7 +3119,7 @@ function initializeTLSOptions(options, servername) {
   options.ALPNProtocols = ['h2'];
   if (options.allowHTTP1 === true)
     ArrayPrototypePush(options.ALPNProtocols, 'http/1.1');
-  if (servername !== undefined && options.servername === undefined)
+  if (servername !== undefined && !options.servername)
```

Comment:
It looks like there was a previous PR that had the same change and @lpinca had left a comment about the behavior.  https://github.com/nodejs/node/pull/39934/files#r699529426

I'm not entirely sure what the desired behavior is here but it's probably worth discussing. In the comment, @lpinca calls out the following commit: https://github.com/nodejs/node/commit/98e9de7db930e505884ca8d0ca6588afbe43f127

Looking at the latest `https` documentation, you can see that `servername` is still documented the same way for `https.Agent`: https://github.com/nodejs/node/blob/9cd2c277d87c6a232f7758966db718fd4b3929de/doc/api/https.md?plain=1#L80

Reply:
Thanks for your reply, I understand the problem much better now, although I think it is still wired we return `https://false`.

Reply:
Is there any conclusion yet on what needs to be done for this?